presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFilter.java

```diff
@@ -52,20 +53,26 @@
     // Mapping from dynamic filter ID to its build channel indices.
     private final Map<String, Integer> buildChannels;
 
-    private final SettableFuture<TupleDomain<Symbol>> resultFuture;
+    // Mapping from dynamic filter ID to its build channel type.
```

Comment:
`LocalDynamicFilter` is created per join node. Therefore `TupleDomain` should be fine here (empty domain means join filters everything)

Reply:
Simplified to make changes only in `LocalDynamicFiltersCollector`